dev-utils/src/commands/init_command.rs

```diff
@@ -0,0 +1,242 @@
+use super::genesis_command::GenesisCommand;
+
+use failure::Error;
+
+use read_input::prelude::*;
+use serde_json::json;
+use serde_json::Value as JsonValue;
+use std::fs;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::process::Command;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct InitCommand {
+    data: String,
+    genesis_time: JsonValue,
+    app_hash: JsonValue,
+    app_state: JsonValue,
+    genesis_dev: JsonValue,
+    tendermint_pubkey: String,
+    staking_account_address: String,
+    distribution_addresses: Vec<String>,
+}
+
+impl InitCommand {
+    pub fn new() -> Self {
+        InitCommand {
+            genesis_time: json!(null),
+            app_hash: json!(null),
+            app_state: json!(null),
+            genesis_dev: json!(null),
+            tendermint_pubkey: "".to_string(),
+            staking_account_address: "".to_string(),
+            distribution_addresses: vec![],
+
+            data: r#"
+        {
+    "distribution": {},
+    "unbonding_period": 60,
+    "required_council_node_stake": "1250000000000000000",
+    "initial_fee_policy": {
+        "base_fee": "1.1",
+        "per_byte_fee": "1.25"
+    },
+    "council_nodes": [
+       
+    ],
+    "launch_incentive_from": "0x35f517cab9a37bc31091c2f155d965af84e0bc85",
+    "launch_incentive_to": "0x20a0bee429d6907e556205ef9d48ab6fe6a55531",
+    "long_term_incentive": "0x71507ee19cbc0c87ff2b5e05d161efe2aac4ee07",
+    "genesis_time": "2019-03-21T02:26:51.366017Z"
+}
+        "#
+            .to_string(),
+        }
+    }
+
+    pub fn read_wallet(&mut self, id: &str, default1: &str, default2: &str) {
+        let obj = self.genesis_dev.as_object_mut().unwrap();
+        let distribution = obj["distribution"].as_object_mut().unwrap();
+        let a = input()
+            .msg(format!("wallet {}({}) address=", id, default1))
+            .default(default1.to_string())
+            .get();
+        let b = input()
+            .msg(format!("wallet {}({}) amount=", id, default2))
+            .default(default2.to_string())
+            .get();
+        distribution.insert(a.to_string(), json!(b));
+        self.distribution_addresses.push(a.to_string());
+    }
+    pub fn read_information(&mut self) {
+        println!("------------{}", self.tendermint_pubkey);
+        self.genesis_dev =
+            serde_json::from_str(&self.data).expect("failed to parse genesis dev config");
```

Comment:
as mentioned above -- I'm not sure why it's parsing a string that's hardcoded in the code.
Wouldn't it be simple to construct the Rust struct directly (+ if the struct changes, it'll be caught by the compiler -- whereas with this hardcoded json string, one would only found out when the code is run)?

Reply:
agree, i'll change using struct

Reply:
it's removed